src/mapping.ts

```diff
@@ -17,7 +18,12 @@ function createPool(poolId: string) : void {
     let poolMeta = poolFromId(poolId);
 
     let seniorTranche = SeniorTranche.bind(<Address>Address.fromHexString(poolMeta.senior))
-    let interestRateResult = seniorTranche.try_ratePerSecond()
+    let assessor_v3 = AssessorV3.bind(<Address>Address.fromHexString(poolMeta.assessor))
+
+    let interestRateResult = poolMeta.version === 3
+      ? assessor_v3.try_seniorInterestRate()
+      : seniorTranche.try_ratePerSecond() 
```

Comment:
The definitions of `seniorTranche.ratePerSecond` is `uint public ratePerSecond;      // interest rate per second in RAD (10^27)` (https://github.com/centrifuge/tinlake/blob/v0.2.5/src/lender/tranche/senior_tranche.sol#L31), while the definition of `assessor_v3.seniorInterestRate` is the `Fixed27` struct (https://github.com/centrifuge/tinlake/blob/develop/src/lender/assessor.sol). So I think that's the same type, right?

Reply:
Maybe, let's just proceed with this and see whether the numbers make sense eventually. It might be that ratePerSecond is the interest rate per second, while seniorInterestRate is an annual rate.